fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImpl.java

```diff
@@ -70,8 +72,10 @@
      * @param serviceConfig  Configuration for the Druid Service
      * @param mapper  A shared jackson object mapper resource
      */
-    public AsyncDruidWebServiceImpl(DruidServiceConfig serviceConfig, ObjectMapper mapper) {
-        this(serviceConfig, initializeWebClient(serviceConfig.getTimeout()), mapper);
+    public AsyncDruidWebServiceImpl(DruidServiceConfig serviceConfig, ObjectMapper mapper,
+            Supplier<Map<String, String>> authHeaders
```

Comment:
Can we re-name this? Rather than auth-specific, can this be named so that it can be _any_ headers?